lib/exirc/utils.ex

```diff
@@ -25,7 +25,7 @@ defmodule ExIrc.Utils do
     end
   end
 
-  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?:(?<user>[^@ ]+)@)?(?:(?<host>[\w.:-]+)))?$/
+  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?:(?<user>[^@ ]+)@)?(?:(?<host>\S+[\w.:-]+)))?$/
```

Comment:
Yeah, that's correct. The current version won't match, for example, `foo/`, and is also slightly less efficient.

In fact, there's also a few redundant non-capturing groups in there.

The whole line should look like the following:

```elixir
  @prefix_pattern ~r/^(?<nick>[^!]+)(?:!(?<user>[^@ ]+@)?(?<host>\S+))?$/
```